utils.py

```diff
@@ -48,6 +51,28 @@ def perform_rmd_conversion(stream):
             out_file.close()
             os.remove(out_path)
         return read_outfile
+    
+# We need to sanitize any code blocks (ex ```{bash}) from the rmd before rendering it
+# This is because kitr (https://rmarkdown.rstudio.com/authoring_quick_tour.html#Rendering_Output.), which powers our rendering
+# Actually executes any code in a labeled codeblock. By processing out any of these labels, we ensure we display a preview without a possible arbitrary code execution vulnerability
+# See this document, issue number one, for more details on the vulnerability: https://docs.google.com/document/d/1aNCOKitTJH-GEkBSR4i-x91O0OQCZ8ZYa3feXtkja94/edit#heading=h.rvpr6zoz0jem 
+# Takes a string and returns binary
+def sanitize_rmd(data):
+    print('printing data to sanitize pre-split', data)
+    lines = data.split('\n')
+    print('printing lines of file', lines)
+    sanitized_file = []
+    for line in lines:
+        if line.startswith('```'):
```

Comment:
my worry here is that code blocks can cover multiple lines, so maybe we sanitize any lines after a line.startswith('```') until a line.endswith('```')

Reply:
I agree with Lucy here, you might miss code otherwise. Testing with like a 3 line code block will let you know for sure

Reply:
Something like this might work to capture everything in-between ```
```suggestion
    code_block = False
    sanitized_data = []
    for line in data:
        if '```' in line:
            sanitized_line = ''
            code_block = not code_block
        elif code_block == True:
            sanitized_line = ''
        else:
            sanitized_line = line
        sanitized_data.append(sanitized_line)
```

Reply:
This is not how knitr works. knitr requires a label of '```{my_lang_here}' at the start of the code block to know the language and to run it in that language. We want to keep the triple backticks, or we will lose the preview of the code as seen in the second screenshot. We are trying to remove anything **after** three backticks, so knitr does not try to run the code within the code block.

In other words, we want to keep all lines, just not allow anything after a triple back tick. A good example can be found in the README of this codebase. If calhoun is provided this .md, We want to remove all language definitions after  the triple back ticks, but  keep the body.

Reply:
ohhh ok, my misunderstanding then, you just need to remove that language defintion, thanks for clarifying!